tools/testrunner/testrunner.go

```diff
@@ -295,23 +167,41 @@ func Main() {
 		log.Fatalf("failed to parse command line options: %v", err)
 	}
 
+	var currentAttempt *attempt
```

Comment:
nit: could you instead extract the last attempt from the runner?

Reply:
I don't quite follow. The `currentAttempt` is assigned to the result of `r.newAttempt()`

Reply:
My point is that if you have a method to get the last attempt from the runner, you can get rid of this mutable variable which takes away from the code's readability.
If `currentAttempt` is always `r.attempts[len(r.attempts)-1]`, there's not much value in doing this booking.